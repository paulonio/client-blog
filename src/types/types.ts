export interface Post {
  username: string;
  date: string;
  title: string;
  body: string;
  urlToImage: string;
}
