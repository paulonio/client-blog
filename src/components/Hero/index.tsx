import React from 'react';

import Link from 'next/link';
import Button from '@/components/Button';

import { sen } from '@/styles/fonts';

import styles from './styled.module.scss';

const Hero = () => {
  return (
    <section className={styles.hero}>
      <div className="container">
        <div className={styles.content}>
          <h4 className={styles.subtitle}>
            POSTED ON <span className={styles.subtitle__bold}>STARTUP</span>
          </h4>
          <h2 className={`${styles.title} ${sen.className}`}>
            Step-by-step guide to choosing great font pairs
          </h2>
          <div className={styles.description__wrapper}>
            <p className={styles.description}>
              By <span className={styles.description__name}>James West</span> | May 23, 2022
            </p>
            <p className={styles.description}>
              Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat
              nulla pariatur. Excepteur sint occaecat cupidatat non proident.
            </p>
          </div>
          <Link href="/blog/post">
            <Button>Read More {`>`}</Button>
          </Link>
        </div>
      </div>
    </section>
  );
};

export default Hero;
