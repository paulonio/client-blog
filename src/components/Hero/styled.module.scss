@import '../../styles/sass/constants';

.hero {
  width: 100vw;
  height: calc(100vh - 80px);
  padding-top: 128px;
  background: radial-gradient(30.56% 76.04% at 74.58% 0%, rgba(0, 0, 0, 0) 0%, rgba(0, 0, 0, 0.6) 100%), url('/assets/images/hero-image.jpg');
  background-size: cover;

  @media screen and (max-width: 768px) {
    padding-top: 0;
    display: flex;
    align-items: center;
    justify-content: center;
  }
}

.content {
  width: 63%;
  max-width: 804px;
  color: $color-white;

  @media screen and (max-width: 768px) {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}

.subtitle {
  font-size: 16px;
  line-height: 20px;
  font-weight: 500;
  color: $color-white;

  &__bold {
    font-weight: 900;
    text-transform: uppercase;
  }
}

.title {
  margin-top: 24px;
  font-size: 56px;
  line-height: 64px;
  font-weight: 700;
  letter-spacing: -2px;
  color: $color-white;

  @media screen and (max-width: 768px) {
    font-size: 44px;
    line-height: 52px;
  }

  @media screen and (max-width: 374px) {
    font-size: 32px;
    line-height: 40px;
  }
}

.description {
  font-size: 16px;
  line-height: 28px;
  font-weight: 400;

  &__wrapper {
    margin: 24px 0 48px 0;
    display: flex;
    flex-direction: column;
    row-gap: 16px;
  }

  &__name {
    color: $color-primary;
  }
}