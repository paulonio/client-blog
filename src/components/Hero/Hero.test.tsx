import React from 'react';
import { render } from '@testing-library/react';
import Hero from '.';
import { PostType } from '@/types/types';

const TEST_POST: PostType = {
  id: 'test',
  body: 'test',
  category: 'test',
  date: 'test',
  tags: ['test'],
  title: 'test',
  urlToAvatar: 'test',
  urlToImage: 'test',
  username: 'test',
};

describe('Hero', () => {
  it('should render hero', () => {
    const { container } = render(<Hero post={TEST_POST} />);

    expect(container).toMatchSnapshot();
  });
});
