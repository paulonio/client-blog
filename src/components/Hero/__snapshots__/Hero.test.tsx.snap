// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should render hero 1`] = `
<div>
  <section
    class="hero"
  >
    <div
      class="container"
    >
      <div
        class="content"
      >
        <h4
          class="subtitle"
        >
          <span
            class="subtitle__bold"
          >
            category
          </span>
        </h4>
        <h2
          class="title className"
        >
          title
        </h2>
        <div
          class="description__wrapper"
        >
          <p
            class="description"
          >
            <span
              class="description__name"
            >
              author
            </span>
             | 
            date
          </p>
          <p
            class="description"
          >
            body
          </p>
        </div>
        <a
          href="/blog/test"
        >
          <button
            class="button"
            type="button"
          >
            read
             
            &gt;
          </button>
        </a>
      </div>
    </div>
  </section>
</div>
`;
